app/models/post.rb

```diff
@@ -0,0 +1,30 @@
+# == Schema Information
+#
+# Table name: posts
+#
+#  id         :bigint           not null, primary key
+#  body       :text(65535)      not null
+#  images     :string(255)      not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  user_id    :bigint
+#
+# Indexes
+#
+#  index_posts_on_user_id  (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => users.id)
+#
+class Post < ApplicationRecord
+  mount_uploaders :images, ImageUploader
+  # 複数の画像を取り扱う場合、serializeメソッドが必要
+  # JSON形式でなくとも、複数の画像を受け取ることは可能
+  # ただ、posts_controllerにてJSON形式でデータを受け取るよう指定しているので、整合性を取る必要あり
+  serialize :images, JSON
```

Comment:
複数の画像を取り扱う場合に限らずserializeが必要なんじゃないかな？

Reply:
[ActiveRecord::AttributeMethods::Serialization::ClassMethods](https://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html)

>"If you have an attribute that needs to be saved to the database as an object, and retrieved as the same object, then specify the name of that attribute using this method and it will be handled automatically. The serialization is done through YAML...."

Railsのアクティブレコードのメソッドみたいですね。。。
日本語だとブログ的な情報しかヒットしないので英語で探しちゃいました。

「オブジェクトとしてDBに保存する属性がある場合は・・・」と書いてあるのですが、要は、JSON形式の場合はそのまま保存できないので、その属性をオブジェクトとして保存するserializeメソッドが必要ということですかね！（自分で書いておいてあまりよく分かっていないですが汗）

となると、複数の画像を取り扱うかは関係なさそうですね。
Qiitaか何かで見た情報を鵜呑みにしちゃいました笑

ちなみに、ポスグレなんかを使っていると、この辺りを自動的にやってくれるらしいです。
> "Keep in mind that database adapters handle certain serialization tasks for you. For instance: json and jsonb types in PostgreSQL will be converted between JSON object/array syntax and Ruby Hash or Array objects transparently. There is no need to use serialize in this case."

MySQLのjson型・・・
MySQLだとJSONのまま保存できるので、serialize不要ってことですかね？

むずかしい・・・

Reply:
そうそう、この辺は明日の定例ではなしましょーか！

Reply:
あー、すみません、単一画像の場合はserialize不要だ。

Reply:
余裕があればぜひお願いします！